core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -204,6 +213,7 @@ case object TaskResultLost extends TaskFailedReason {
 @DeveloperApi
 case object TaskKilled extends TaskFailedReason {
   override def toErrorString: String = "TaskKilled (killed intentionally)"
+  override val countTowardsTaskFailures: Boolean = false
```

Comment:
the switch to a `val` came from an earlier discussion with @kayousterhout ... there was some other confusion, thought maybe changing to a val would make it more clear it is a constant.  But I don't think either of feels strongly, the argument to switch to a val was pretty weak.  I can change it back